Sources/TuistSupport/Utils/Glob.swift

```diff
@@ -113,15 +113,37 @@ public class Glob: Collection {
     }
 
     private func expandGlobstar(pattern: String) -> [String] {
-        guard pattern.contains("**") else {
+        // Split pattern string by slash to find globstar.
+        let patternComponents = pattern.components(separatedBy: "/")
+
+        // We are only interested in the first globstar since that is where we want to separate the pattern string.
+        guard let pivot = patternComponents.firstIndex(of: "**") else {
             return [pattern]
         }
 
         var results = [String]()
-        var parts = pattern.components(separatedBy: "**")
-        let firstPart = parts.removeFirst()
-        var lastPart = parts.joined(separator: "**")
 
+        // Part before the first globstar
+        let firstPartLowerBound = 0
+        let firstPartUpperBound = pivot
+        let firstPartComponents: ArraySlice<String> = if firstPartLowerBound < firstPartUpperBound {
+            patternComponents[firstPartLowerBound ..< firstPartUpperBound]
+        } else {
+            []
+        }
+        let firstPart = firstPartComponents.joined(separator: "/")
+
+        // Part after the first globstar
+        let lastPartLowerBound = pivot + 1
+        let lastPartUpperBound = patternComponents.count
+        let lastPartComponents: ArraySlice<String> = if lastPartLowerBound < lastPartUpperBound {
+            patternComponents[lastPartLowerBound ..< lastPartUpperBound]
+        } else {
+            []
+        }
```

Comment:
Why not use the `substring` methods instead of the String subscript accessors?

Reply:
`patternCompotents` is not a `String` but an array of `String`. 
We first separate the pattern string by `/`, divide them into two groups by the first globstar, and finally join each groups back with `/`.
By doing this, we can split pattern string into two parts. The first part before the globstar and the other part after the globstar.

Reply:
The globstars behind will be expanded by recursive calls at the end of the function.

Reply:
gotcha, that's fine then, thanks for clarifying